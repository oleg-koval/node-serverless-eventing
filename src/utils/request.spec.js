const { parseWith } = require('./request')

describe('request data parser', () => {
    describe('with the parser', () => {
        const req = parseWith(JSON.parse)
        it('should work', () => {
            expect(
                req({
                    a: 1,
                })
            ).toMatchSnapshot()
        })
        it('no data', () => {
            expect(() => req()).toThrowErrorMatchingSnapshot()
        })
        it('valid JSON string', () => {
            expect(req('{"a": 1}')).toMatchSnapshot()
        })
    })

    describe('without the parser', () => {
        it('should throw an error', () => {
            expect(() => parseWith()('{"a": 1}')).toThrowErrorMatchingSnapshot()
        })
    })
})
