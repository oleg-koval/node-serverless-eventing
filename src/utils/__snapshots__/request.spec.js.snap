// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`request data parser with the parser no data 1`] = `"data missing"`;

exports[`request data parser with the parser should work 1`] = `
Object {
  "a": 1,
}
`;

exports[`request data parser with the parser valid JSON string 1`] = `
Object {
  "a": 1,
}
`;

exports[`request data parser without the parser should throw an error 1`] = `"parser missing"`;
